api/src/db/models/user_models.py

```diff
@@ -244,3 +248,18 @@ class SuppressedEmail(ApiSchemaTable, TimestampMixin):
     email: Mapped[str] = mapped_column(index=True)
     reason: Mapped[str]
     last_update_time: Mapped[datetime] = mapped_column(index=True)
+
+
+class UserApiKey(ApiSchemaTable, TimestampMixin):
+    """API Key table for user authentication to the API"""
+
+    __tablename__ = "user_api_key"
+
+    token_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True, default=uuid.uuid4)
+    key_name: Mapped[str] = mapped_column(nullable=False)
+    key_id: Mapped[str] = mapped_column(nullable=False, comment="AWS API Gateway key identifier")
+    user_id: Mapped[uuid.UUID] = mapped_column(UUID, ForeignKey("api.user.user_id"), index=True)
```

Comment:
Having to use a string in the ForeignKey bit is only needed if there's a circular dependency of the types - it works the same either way
```suggestion
    user_id: Mapped[uuid.UUID] = mapped_column(UUID, ForeignKey(User.user_id), index=True)
```